bundle/src/main/java/com/adobe/acs/commons/mcp/processes/AssetReport.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    private String getParentPath(String path) {
+        return path.substring(0, path.lastIndexOf('/'));
+    }
+
+    private void setValue(String path, Column field, long amount) {
+        if (getDepth(path) < depthLimit) {
+            synchronized (report) {
+                EnumMap<Column, Long> row = getReportRow(path);
+                row.put(field, amount);
+            }
+        }
+    }
+
+    private void tabulate(String path, Column counter, long amount) {
+        if (getDepth(path) < depthLimit && path.length() >= baseFolder.length()) {
+            synchronized (report) {
+                EnumMap<Column, Long> row = getReportRow(path);
+                if (row.containsKey(counter)) {
+                    row.put(counter, row.get(counter) + amount);
+                } else {
+                    row.put(counter, amount);
+                }
+            }
+        }
+        if (path.length() > baseFolder.length()) {
+            tabulate(getParentPath(path), counter, amount);
+        }
+    }
+
+    private void examineAsset(ResourceResolver rr, String assetPath) throws RepositoryException, Exception {
+        Actions.setCurrentItem(assetPath);
+        String folderPath = getParentPath(assetPath);
+        Set<String> observedHashes = new HashSet<>();
+
+        Asset asset = rr.getResource(assetPath).adaptTo(Asset.class);
+        String hash = asset.getMetadataValue(SHA1);
+        if (hash != null) {
+            observedHashes.add(hash);
+        }
+        Map<String, Rendition> renditions = new HashMap<>();
+        asset.listRenditions().forEachRemaining(r -> renditions.put(r.getName(), r));
+        Rendition original = renditions.remove("original");
+        tabulate(folderPath, Column.rendition_count, renditions.size());
+
+        if (original != null) {
+            long size = original.getSize();
+            tabulate(folderPath, Column.original_size, size);
+            tabulate(folderPath, Column.combined_size, size);
+        }
+
+        renditions.values().forEach(rendition -> {
+            long size = rendition.getSize();
+            tabulate(folderPath, Column.rendition_size, size);
+            tabulate(folderPath, Column.combined_size, size);
+        });
+
+        if (includeSubassets) {
+            DamUtil.getSubAssets(asset.adaptTo(Resource.class)).stream().forEach(subasset -> {
+                tabulate(folderPath, Column.subasset_count, 1);
+                long size = subasset.getRenditions().stream().collect(Collectors.summingLong(Rendition::getSize));
+                tabulate(folderPath, Column.subasset_size, size);
+                tabulate(folderPath, Column.combined_size, size);
+            });
+        }
+        if (includeVersions) {
+            for (Revision rev : asset.getRevisions(null)) {
+                tabulate(folderPath, Column.version_count, 1);
+                FrozenAsset assetVersion = new FrozenAsset(asset, rev);
```

Comment:
I adopted this approach because Clint already did the hard part of implementing this, so it seemed the shortest path.  My original approach was to walk the tree the hard way.  Yes the overall goal is to count the size of binaries both for the versioned asset's renditions as well as its versioned subassets.  When you consider how much stuff is being stored in a version you'll quickly see why keeping too many versions is a very bad thing.